src/eth/evm/revm.rs

```diff
@@ -174,7 +182,18 @@ impl Database for RevmDatabaseSession {
                 .insert(account.address.clone(), ExecutionAccountChanges::from_existing_account(account.clone()));
         }
 
-        Ok(Some(account.into()))
+        let bytecode = if let Some(bytes) = account.bytecode {
+            Some(self.bytecodes.entry(address).or_insert(to_analysed(bytes.into())).clone())
```

Comment:
You cannot cache the bytecode until you are sure the transaction is confirmed, otherwise you can have an invalid bytecode associated with an address.

This is the reason I populated the cache only after the transaction is finished and confirmed to be a success.